src/database/stores/RethinkDbGuildStore.ts

```diff
@@ -788,13 +791,59 @@ export class RethinkDbGuildStore implements GuildStore {
     }
 
     public async migrate(): Promise<void> {
-        const indexes = await this.#table.query(t => t.indexList());
-        if (!indexes.includes('interval')) {
-            await this.#table.query(t => t.indexCreate('interval', r => r('ccommands').hasFields('_interval')));
+        const versionObj = await this.#table.get('migration', true);
+        if (versionObj === undefined) {
+            await this.#table.insert({
+                active: false,
+                ccommands: {},
+                channels: {},
+                guildid: 'migration',
+                name: '0',
+                settings: {}
+            });
+        }
+        let currentVersion = parseInt(versionObj?.name ?? '0');
+        for (const migration of migrations.filter(v => v.id > currentVersion).sort((a, b) => a.id - b.id)) {
+            this.#logger.database('Migrating GuildStore from', currentVersion, 'to', migration.id);
+            await migration.migrate(this.#table);
+            await this.#table.update('migration', { name: migration.id.toString() });
+            this.#logger.database('Migrated GuildStore from', currentVersion, 'to', migration.id);
+            currentVersion = migration.id;
         }
     }
 }
 
+interface GuildStoreMigration {
+    readonly id: number;
+    migrate(table: RethinkDbTable<StoredGuild>): Promise<void>;
+}
+
+const migrations: GuildStoreMigration[] = [
+    {
+        id: 1,
+        async migrate(table) {
+            const indexes = await table.query(t => t.indexList());
+            if (!indexes.includes('interval')) {
+                await table.query(t => t.indexCreate('interval', r => r('ccommands').hasFields('_interval')));
+            }
+        }
+    },
+    {
+        id: 2,
+        async migrate(table) {
+            for await (const guild of table.stream(r => r.filter(g => g('ccommands').values().map(cmd => cmd.hasFields('id')).contains(false)))) {
+                const ccommands: Record<string, { id: string; }> = {};
+                for (const [name, ccommand] of Object.entries(guild.ccommands)) {
+                    if (ccommand?.id === undefined) {
+                        ccommands[name] = { id: snowflake.create().toString() };
+                    }
+                }
+                await table.update(guild.guildid, { ccommands });
+            }
+        }
+    }
+];
+
```

Comment:
wouldn't it be better to structure this? we might need it more than once (actually it's the second time we run a migration so a third is probably not avoidable in the future

Reply:
I think ill work on a proper system for doing migrations after this, I think doing it in this PR would be a bit of scope creep. This was the bare minimum needed for this to become functional, so for now ill leave it as this.